File: algorithms/Arrays/Easy/Anagrams/VSDAnagrams.py
#  Copyrights to venkys.io
#  For more programs visit venkys.io 
#  Python program for Anagrams



def is_anagram(s1,s2):
    # we use hashtable to solve this problem
    if len(s1) != len(s2):
        print("The strings ",s1,s2,"are not anagrams")
    else:
        ht=dict()
        for i in s1:
            if i in ht.keys():
                ht[i]+=1
            else:
                ht[i]=1
        for j in s2:
            if j not in ht.keys():
                return False
            ht[j]-=1
        for key,value in ht.items():
            if value!=0:
                return False 
        return True

                
# test drive code
if __name__ == "__main__":
    s1="hello"
    s2="olleh"
    if(is_anagram(s1,s2)):
        print(f"{s1} and {s2} are Anagrams")
    else:
        print(f"{s1} and {s2} are Not Anagrams")

File: algorithms/Arrays/Easy/Anagrams/VSDAnagrams.cpp
// Copyrights to venkys.io
// For more programs visit venkys.io 
// CPP program for Anagrams


#include<iostream>
#include<bits/stdc++.h>
using namespace std;

int main(){
  string s1="hello";
  string s2="olleh";
  
  if(s1.size()!=s2.size()){
    cout<<"Not Anagram";
  }
  else{
    map<char,int> map;
    for(int i=0;i<s1.size();i++){
      map[s1[i]]++;
    }
    for(int j=0;j<s2.size();j++){
      if(map.find(s2[j])!=map.end()){
        map[s2[j]]--;
      }
      else{
        cout<<s1<<" and "<<s2<<" are Not Anagrams";
        return false;
      }
    }
    for(auto items:map){
      if(items.second!=0){
         cout<<s1<<" and "<<s2<<" are Not Anagrams";
      }
    }
    cout<<s1<<" and "<<s2<<" are Anagrams";
    return true;
  }
   
  return 0;
}